source/secure_tunneling.c

```diff
@@ -2423,28 +2420,60 @@ struct aws_secure_tunnel *aws_secure_tunnel_new(
         goto error;
     }
 
+    secure_tunnel->host_resolution_config = aws_host_resolver_init_default_resolution_config();
+    secure_tunnel->host_resolution_config.resolve_frequency_ns =
+        aws_timestamp_convert(MAX_RECONNECT_DELAY_MS, AWS_TIMESTAMP_MILLIS, AWS_TIMESTAMP_NANOS, NULL);
+
     secure_tunnel->desired_state = AWS_STS_STOPPED;
     secure_tunnel->current_state = AWS_STS_STOPPED;
 
-    if (options->root_ca != NULL) {
-        if (aws_tls_ctx_options_override_default_trust_store_from_path(&tls_ctx_opt, NULL, options->root_ca)) {
+    /* tls setup */
+    if (options->tls_options) {
+        if (aws_tls_connection_options_copy(&secure_tunnel->tls_con_opt, options->tls_options)) {
```

Comment:
Should we always override the root_ca file as long as user set it up? Or at least we should mention that the `root_ca` might be ignored in the certain case? Sorry I'm not entirely sure how does the tls_options setup.

Reply:
My understanding is that if tls_options are passed in, the expectation is that an overriden root_ca should already have been applied to it. The only change I'm making to an already setup aws_tls_connection_options will be the endpoint applied to the copy we keep.